Automattic/jetpack
# Connection: allow provisioning using an application password

## Proposed changes:
* Modify permission check for `/jetpack/v4/remote_provision` endpoint to allow application passwords.

### Other information:


## Jetpack product discussion
https://github.com/Automattic/vulcan/issues/620

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Connect Jetpack in site-only mode.
2. Go to "Users -> Profile" and create an application password.
3. Run the cURL command using your username and application password:
```bash
curl 'https://example.com/index.php?rest_route=/jetpack/v4/remote_provision' -X POST -H 'Content-Type: application/json' -u 'username:application-password'
```
4. The response should contain many fields, among which `secret` (random string) and `user_id` (matching your WP user ID).
5. Run the same command without `-u 'username:application-password'`. Confirm you get the `invalid_permission_remote_provision` error.
